index.js

```diff
@@ -8,19 +8,22 @@ var validate = require('./lib/validate');
   * Unpacks, parses, validates, and analyzes Scratch projects. If successful,
   * will return a valid Scratch project object with appended metadata.
   * @param {Buffer | string} input    Buffer or string representing project
+  * @param {boolean}         isSprite Whether this is a sprite (true) or whole project (false)
   * @param {Function}        callback Returns error or project data
   */
-module.exports = function (input, callback) {
+module.exports = function (input, isSprite, callback) {
```

Comment:
Because we are changing the interface in a breaking way it would be good to either:
- Do a major version bump on the package.json (breaking change)
- Check to see if `callback` is `undefined` and then default and "compress" the args to maintain compatibility

My preference would be for the second option, but I'm totally flexible.

Reply:
I think the way I've committed, I've already made it so that this would be a breaking change (it's in the first commit message), but I'm happy to amend the commit message.

I like the second option, but wasn't sure if there was a good way to introduce a new argument before the callback without making a breaking change. I'm not sure if I completely understand your suggestion above.

I'm guessing you're saying do something like this:

```
if ((typeof callback === 'undefined') && (typeof isSprite === 'Function')) {
    // assume that the user is using old version of API and callback was provided as second argument instead of third
    callback = isSprite;
    isSprite = false;
}
```

I'm personally not a fan of the above because the API has changed but we're pretending it hasn't, so anyone reading just the documentation of this code might get confused reading a usage of this function in the wild, but I'm happy to change it if we prefer not to make breaking changes.

Reply:
Yup. That is what I was suggesting. I'm ok to proceed as-is if you prefer.

Reply:
Just thought of another idea -- What if I just provide 2 APIs and keep the old one as the default export? I think that's a thing we can do right?

E.g. `validateProject` becomes default, but there's also `validateSprite`